src/services/inplayer.account.service.ts

```diff
@@ -328,3 +333,75 @@ function parseJson(value: string, fallback = {}) {
 export const canUpdateEmail = false;
 
 export const canChangePasswordWithOldPassword = true;
+
+export const initCustomerExtras = async (): Promise<ExternalData> => {
+  const [favoritesData, historyData] = await Promise.all([InPlayer.Account.getFavorites(), await InPlayer.Account.getWatchHistory({})]);
+
+  const favorites = favoritesData.data?.collection?.map((favorite: FavoritesData) => {
+    return processFavorite(favorite);
+  });
+
+  const history = historyData.data?.collection?.map((history: WatchlistHistory) => {
+    return processHistoryItem(history);
+  });
+
+  return {
+    favorites,
+    history,
+  };
+};
+
+export const updatePersonalShelves: UpdatePersonalShelves = async (payload) => {
+  const { favorites, history } = payload.externalData;
+  const user = useAccountStore.getState().user;
+  const currentFavoriteIds = user?.externalData?.favorites?.flatMap((e) => e.mediaid);
+  const payloadFavoriteIds = favorites?.flatMap((e) => e.mediaid);
+
+  try {
+    history.forEach(async (history) => {
+      if (user?.externalData?.history?.length) {
```

Comment:
I don't understand what's happening here. Can you add some commenting (unless we end up refactoring it)

Reply:
this is a universal and InPlayer-specific method for converting an array of favorites/watch-history items into separate requests to the InPlayer API.

Because we do not support bulk inserts( like cleeng ), in this method, we calculate the delta( the difference between the local arrays and remote arrays) of the arrays from the local storage/account store.

Every time when the array is changed( when the users click on the favorite button, the current flow), we send separate requests for each favorite/watch-history item from the 'delta' to the InPlayer's API.

For this part, for the watch history, we always send a patch request to update/create the watch-history item for the user.

This method is also used to sync the data from the local storage when the user is newly registered or logged in.